src/solvers/dg_curved/dg_2d.jl

```diff
@@ -78,7 +78,7 @@ end
 
 function calc_interface_flux!(cache, u,
                               mesh::CurvedMesh{2},
-                              equations, dg::DG)
+                              equations, surface_integral, dg::DG)
```

Comment:
What I am wondering here (just came to my mind): Shouldn't we wait with the introduction of all this dispatch machinery virtually everywhere until it is actually required? I am asking since this seems to be an infrastructural change that is not really required yet and that might be better suited to be introduced when it's needed because there are additional methods with different dispatch types. What is your motivation/take on this?

Reply:
I wanted to keep my FD PR smaller. If you're okay with it, I can also just push everything here. Since you've already seen this part, this is probably the way to go.